pyccel/ast/core.py

```diff
@@ -2468,6 +2465,17 @@ def _eval_is_positive(self):
         #we do this inorder to infere the type of Pow expression correctly
         return self.is_real
 
+    def __getitem__(self, *args):
+
+        if len(args) == 1 and isinstance(args[0], (Tuple, tuple, list)):
+            args = args[0]
+
+        if self.rank != len(args):
```

Comment:
you can have something like `x[1]` where the rank of `x` is 2 but you can't have something like `x[1,2,3]`

Reply:
So you would suggest: `if self.rank > len(args)`?

In pyccel we always add `Slice(None,None)` when indexing so this condition is always satisfied. I think probably helps so a Variable is either fully indexed or not indexed at all. Do you think this handling is likely to change? @saniQQ does the C implementation of slicing require us to provide slices for all dimensions?

Reply:
yes we should only have `if self.rank > len(args)`, I don't know how it's treated in the C language, but the missing indices should automaticaly added here

Reply:
Currently the extra indexes are added in the `SemanticParser`. Do you want me to move this to the constructor of `IndexedElement`? I would rather avoid doing it in the `__getitem__` method as this will lead to code duplication

Reply:
ah okay, we probably need to do it the constructor

Reply:
done